x-pack/platform/plugins/shared/fleet/server/services/agents/crud.ts

```diff
@@ -395,6 +387,99 @@ export async function getAllAgentsByKuery(
   };
 }
 
+/**
+ * Fetch all agents by kuery in batches.
+ * @param esClient
+ * @param soClient
+ * @param options
+ */
+export async function getTotalAgentsByKuery(
```

Comment:
We could rename this to `getTotalAgentsCountByKuery`, it was not immediately obvious that it returns the count only, not the actual agents.